src/chips/opn_chip_base.tcc

```diff
@@ -5,6 +5,18 @@
 #include <zita-resampler/vresampler.h>
 #endif
 
+#if !defined(LIKELY) && defined(__GNUC__)
+#define LIKELY(x) __builtin_expect((x), 1)
+#elif !defined(LIKELY)
+#define LIKELY(x) (x)
+#endif
+
+#if !defined(UNLIKELY) && defined(__GNUC__)
+#define UNLIKELY(x) __builtin_expect((x), 0)
+#elif !defined(UNLIKELY)
+#define UNLIKELY(x) (x)
```

Comment:
Is this correct? I think "LIKELY" must do `x == 1` and `UNLIKELY` must do `x == 0`?
Even look the simple demo:
```c
#include <stdio.h>

int main()
{
    printf("Likely   1 == 1 >> %li\n", __builtin_expect(1, 1));
    printf("Likely   0 == 1 >> %li\n", __builtin_expect(0, 1));
    printf("UnLikely 1 != 0 >> %li\n", __builtin_expect(1, 0));
    printf("UnLikely 0 != 0 >> %li\n", __builtin_expect(0, 0));
    return 0;
}
```
and it's output:
```
$ ./likelyunlikely 
Likely   1 == 1 >> 1
Likely   0 == 1 >> 0
UnLikely 1 != 0 >> 1
UnLikely 0 != 0 >> 0
```

Reply:
Anyway, I have extended my demo:
```c
#include <stdio.h>

#define xxLIKELY(x) (x)
#define xxUNLIKELY(x) (x)

int main()
{
    printf("Likely   1 == 1 >> %li vs %d\n", __builtin_expect(1, 1), xxLIKELY(1));
    printf("Likely   0 == 1 >> %li vs %d\n", __builtin_expect(0, 1), xxLIKELY(0));
    printf("UnLikely 1 != 0 >> %li vs %d\n", __builtin_expect(1, 0), xxUNLIKELY(1));
    printf("UnLikely 0 != 0 >> %li vs %d\n", __builtin_expect(0, 0), xxUNLIKELY(0));
    printf("Likely   2 == 1 >> %li vs %d\n", __builtin_expect(2, 1), xxLIKELY(2));
    printf("Likely   1 == 1 >> %li vs %d\n", __builtin_expect(1, 1), xxLIKELY(1));
    printf("Likely   2 == 1 >> %li vs %d\n", __builtin_expect(2, 0), xxUNLIKELY(2));
    printf("Likely   1 == 1 >> %li vs %d\n", __builtin_expect(1, 0), xxUNLIKELY(1));
    return 0;
}
```
And seems must be ok, but just looks weird...
```
$ ./likelyunlikely 
Likely   1 == 1 >> 1 vs 1
Likely   0 == 1 >> 0 vs 0
UnLikely 1 != 0 >> 1 vs 1
UnLikely 0 != 0 >> 0 vs 0
Likely   2 == 1 >> 2 vs 2
Likely   1 == 1 >> 1 vs 1
Likely   2 == 1 >> 2 vs 2
Likely   1 == 1 >> 1 vs 1
```

Reply:
I think you misunderstand `__builtin_expect`. It's not comparison.